src/backend/core/management/commands/reindex_with_embedding.py

```diff
@@ -0,0 +1,171 @@
+        """
+
+        success, message = check_index_dimensions_match(
+            self.opensearch_client_, source_index, destination_index
+        )
+        if not success:
+            raise CommandError(message)
+
+        sample_size = 100
+        source_sample_documents = self.opensearch_client_.search(
+            index=source_index, size=sample_size, body={"query": {"match_all": {}}}
+        )
+        destination_sample_documents = self.opensearch_client_.search(
+            index=destination_index, size=sample_size, body={"query": {"match_all": {}}}
+        )
+
+        success, message = check_embeddings_populated(destination_sample_documents)
+        if not success:
+            raise CommandError(message)
+
+        success, message = check_document_fields(
+            source_sample_documents, destination_sample_documents
+        )
+        if not success:
+            raise CommandError(message)
+
+    def replace_source_index_with_destination_index(
+        self, source_index, destination_index
+    ):
+        """Delete source index and rename destination index to source index."""
+        self.opensearch_client_.indices.delete(index=source_index)
+        self.opensearch_client_.indices.put_alias(
+            index=destination_index, name=source_index
+        )
+
+
+def reindex_with_embedding(
+    opensearch_client_, source_index, destination_index, batch_size=500
+):
+    """Reindex documents from source index to destination index with embeddings."""
+
+    destination_index = f"{source_index}-embedded"
+    ensure_index_exists(destination_index)
+
+    page = opensearch_client_.search(
+        index=source_index,
+        scroll="5m",
+        size=batch_size,
+        body={"query": {"match_all": {}}},
+    )
+    scroll_id = page["_scroll_id"]
+    scroll_size = len(page["hits"]["hits"])
+
+    while scroll_size > 0:
+        actions = []
+        for hit in page["hits"]["hits"]:
+            document = hit["_source"]
+
+            if document.get("embedding") is None:
+                document["embedding"] = embed_text(
+                    f"<{document.get('text')}>:<{document.get('content')}>"
+                )
+
+            actions.append({"index": {"_id": hit["_id"]}})
+            actions.append(document)
+
+        page = opensearch_client_.scroll(scroll_id=scroll_id, scroll="5m")
+        scroll_id = page["_scroll_id"]
+        scroll_size = len(page["hits"]["hits"])
+
+        opensearch_client_.bulk(index=destination_index, body=actions)
+
+    opensearch_client_.clear_scroll(scroll_id=scroll_id)
+
+
+def check_index_dimensions_match(opensearch_client_, source_index, destination_index):
+    """Check that source and destination index have the same document count."""
+    source_count = opensearch_client_.count(index=source_index)["count"]
+    destination_count = opensearch_client_.count(index=destination_index)["count"]
+    if source_count != destination_count:
```

Comment:
As say before, I think it will probably never be the case, except if you run the command during the night ^^